### PR TITLE
fu-tool: Allow to request immediate actions from the end user

### PR DESCRIPTION
**Pull request by gportay**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

This addresses #4658.

### CODE REVIEW

File: src/fu-tool.c [hunk 1]
```diff
@@ -926,6 +932,12 @@ fu_util_update_device_changed_cb(FwupdClient *client, FwupdDevice *device, FuUti
 static void
 fu_util_display_current_message(FuUtilPrivate *priv)
 {
+	/* print all POST requests */
+	for (guint i = 0; i < priv->post_requests->len; i++) {
+		FwupdRequest *request = g_ptr_array_index(priv->post_requests, i);
+		g_print("%s\n", fwupd_request_get_message(request));
+	}
+
 	if (priv->current_message == NULL)
```
Comment (gportay) [inline src/fu-tool.c:941]: I am not sure what to do with the `current_message`; should we drop it?

Comment (hughsie) [reply src/fu-tool.c:941]: Yup, I think so; I think the behaviour between fu-util.c and fu-tool.c should be exactly the same.

Comment (gportay) [reply src/fu-tool.c:941]: Also, what is the purpose of `fu-util-common.c`? if the function is copied from `src/fu-util.c` to `src/fu-tool.c` and it is strickly identical, it should be moved and shared to `fu-util-common.c`, right?

Comment (hughsie) [reply src/fu-tool.c:941]: > it should be moved and shared to fu-util-common.c, right

Ya, but only if it's general purpose. Don't go overboard.

**Comment by gportay** [comment]
Here is the new output (`FWUPD_REQUEST_KIND_IMMEDIATE`):

```
sudo fwupdtool install SteelSeries_Rival_3_Wireless_1.0.cab --allow-older --allow-reinstall 
Loading…                 [-                                      ]
Action Required: Rival 3 Wireless needs to be connected to start the update. Please put the switch button underneath to 2.4G, or click on any button to reconnect it.
Action Required: Rival 3 Wireless needs to be manually restarted to complete the update. Please unplug the 2.4G USB Wireless adapter and then re-plug it.
Waiting…                 [************************************** ]
```

```
sudo fwupdmgr local-install SteelSeries_Rival_3_Wireless_1.0.cab --allow-older --allow-reinstall 
Action Required: Rival 3 Wireless needs to be connected to start the update. Please put the switch button underneath to 2.4G, or click on any button to reconnect it.
Action Required: Rival 3 Wireless needs to be manually restarted to complete the update. Please unplug the 2.4G USB Wireless adapter and then re-plug it.
Waiting…                 [************************************** ]Successfully installed firmware
```

**Comment by gportay** [comment]
Identical, except for the ending `Successfully installed firmware`.

**Review by hughsie — APPROVED** [review]
Looks great, thanks!

**Comment by gportay** [comment]
And here with a `FWUPD_REQUEST_KIND_POST`:

```
sudo fwupdtool install-blob mouse.bin-1.81 4f3fac7c-981b-58cc-bb58-8a3b90f31c9c 
Loading…                 [-                                      ]
Selected device: Aerox 3 Wireless Mouse
Restarting device…       [*******************************        ]12:18:07:0250 FuPluginSteelSeries  failed to reset: failed to do control transfer: transfer failed
Waiting…                 [************************************** ]Aerox 3 Wireless Mouse needs to be manually restarted to complete the update. Please unplug the USB cable and then re-plug it.
```

with this local change:

``` diff
diff --git a/plugins/steelseries/fu-steelseries-fizz.c b/plugins/steelseries/fu-steelseries-fizz.c
index 58a69e76e..5e37f903f 100644
--- a/plugins/steelseries/fu-steelseries-fizz.c
+++ b/plugins/steelseries/fu-steelseries-fizz.c
@@ -516,6 +516,27 @@ fu_steelseries_fizz_attach(FuDevice *device, FuProgress *progress, GError **erro

        fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);

+       /*
+        * the user has to do something
+        *
+        * Note: the device is not reenumerated after a second reset. Ask the user to manually
+        * replug the device to allow successive updates.
+        */
+       /* it is not a USB receiver */
+       if (!fu_device_has_private_flag(device, FU_STEELSERIES_DEVICE_FLAG_IS_RECEIVER)) {
+               g_autoptr(FwupdRequest) request = NULL;
+               g_autofree gchar *msg = NULL;
+
+               msg = g_strdup_printf("%s needs to be manually restarted to complete the update. "
+                                     "Please unplug the USB cable and then re-plug it.",
+                                     fu_device_get_name(device));
+               request = fwupd_request_new();
+               fwupd_request_set_kind(request, FWUPD_REQUEST_KIND_POST);
+               fwupd_request_set_id(request, FWUPD_REQUEST_ID_PRESS_UNLOCK);
+               fwupd_request_set_message(request, msg);
+               fu_device_emit_request(device, request);
+       }
+
        /* success */
        return TRUE;
 }
```

**Pull request merged**